From 83c1851afdd413e7e81b93333992f1127a558ffc Mon Sep 17 00:00:00 2001
From: John Kacur <jkacur@redhat.com>
Date: Tue, 21 Mar 2017 17:11:53 +0100
Subject: [PATCH 129/134] trace-cmd: Fix segmentation fault in trace-snapshot

To reproduce the problem:

  plugin 'wakeup'
trace-cmd: Device or resource busy
  Segmentation fault (core dumped)

When the user triggers a condition such as EBUSY the program should die
gracefully. The problem here is simply caused by an extra conversion
specifier in "die" in the write_file function

Link: http://lkml.kernel.org/r/1490112713-9456-1-git-send-email-jkacur@redhat.com

Signed-off-by: John Kacur <jkacur@redhat.com>
Signed-off-by: Steven Rostedt (VMware) <rostedt@goodmis.org>
---
 trace-snapshot.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/trace-snapshot.c b/trace-snapshot.c
index 9d3608c49563..771b065ccfa4 100644
--- a/trace-snapshot.c
+++ b/trace-snapshot.c
@@ -42,7 +42,8 @@ static void write_file(const char *name, char *val)
 
 	n = write(fd, val, strlen(val));
 	if (n < 0)
-		die("failed to write '%d' to %s\n", path);
+		die("failed to write to %s\n", path);
+
 	tracecmd_put_tracing_file(path);
 	close(fd);
 }
-- 
2.5.5

